src/snarky.d.ts

```diff
@@ -348,6 +348,17 @@ declare const Snarky: {
     ): void;
 
     zero(in1: FieldVar, in2: FieldVar, out: FieldVar): void;
+
+    generic(
+      sl: FieldConst,
+      l: FieldVar,
+      sr: FieldConst,
+      r: FieldVar,
+      so: FieldConst,
+      o: FieldVar,
+      sm: FieldConst,
+      sc: FieldConst
+    ): void;
```

Comment:
I made the generic interface more friendly in TS and left the OCaml side as is (so that everything matches `basic`).

Let me know if you would like me to change anything more (e.g., have [OCaml](https://github.com/o1-labs/o1js-bindings/blob/bc3abc3d583517bba00e4ea17a0226bd5176dfb1/ocaml/lib/snarky_bindings.ml#L117) use the same interface, or just rename these variables).

Reply:
I think it's perfect!